File: frontend/lib/models/clinical_note.dart
class ClinicalNote {
  final int? id;
  final int patientId;
  final int doctorId;
  final String note;

  ClinicalNote({
    this.id,
    required this.patientId,
    required this.doctorId,
    required this.note,
  });

  factory ClinicalNote.fromJson(Map<String, dynamic> json) {
    return ClinicalNote(
      id: json["id"],
      patientId: json["patient_id"],
      doctorId: json["doctor_id"],
      note: json["note"] ?? "",
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "patient_id": patientId,
      "doctor_id": doctorId,
      "note": note,
    };
  }
}

File: backend/app/models/clinical_note.py
from sqlalchemy import Column, Integer, Text, DateTime, ForeignKey
from sqlalchemy.sql import func

from app.database import Base


class ClinicalNote(Base):
    __tablename__ = "clinical_notes"

    id = Column(Integer, primary_key=True, index=True)

    patient_id = Column(
        Integer,
        ForeignKey("patients.id")
    )

    doctor_id = Column(
        Integer,
        ForeignKey("users.id")
    )

    note = Column(Text, nullable=False)

    created_at = Column(
        DateTime(timezone=True),
        server_default=func.now()
    )